test/new-e2e/systemProbe/systemProbeTestEnv.go

```diff
@@ -0,0 +1,135 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package systemProbe
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/utils/credentials"
+	"github.com/DataDog/datadog-agent/test/new-e2e/utils/infra"
+	"github.com/DataDog/test-infra-definitions/aws"
+	"github.com/DataDog/test-infra-definitions/aws/scenarios/microVMs/microvms"
+	"github.com/DataDog/test-infra-definitions/command"
+
+	"github.com/pulumi/pulumi-command/sdk/go/command/remote"
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+type TestEnv struct {
+	context context.Context
+	envName string
+	name    string
+
+	ARM64InstanceIP  string
+	X86_64InstanceIP string
+	StackOutput      auto.UpResult
+}
+
+var (
+	SSHKeyFile           = filepath.Join(".", "/", "aws-ssh-key")
+	vmConfig             = filepath.Join(".", "systemProbe", "config", "vmconfig.json")
+	DD_AGENT_TESTING_DIR = os.Getenv("DD_AGENT_TESTING_DIR")
+	sshKeyX86            = os.Getenv("LibvirtSSHKeyX86")
+	sshKeyArm            = os.Getenv("LibvirtSSHKeyARM")
+)
+
+func NewTestEnv(name, securityGroups, subnets, x86InstanceType, armInstanceType string) (*TestEnv, error) {
+	systemProbeTestEnv := &TestEnv{
+		context: context.Background(),
+		envName: "aws/sandbox",
+		name:    fmt.Sprintf("microvm-scenario-%s", name),
+	}
+
+	awsManager := credentials.NewManager()
+	sshkey, err := awsManager.GetCredential(credentials.AWSSSMStore, "ci.datadog-agent.aws_ec2_kitchen_ssh_key")
+	if err != nil {
+		return nil, err
+	}
+
+	// Write ssh key to file
+	f, err := os.Create(SSHKeyFile)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	f.WriteString(sshkey)
+
+	stackManager := infra.GetStackManager()
+
+	config := auto.ConfigMap{
+		"ddinfra:aws/defaultARMInstanceType": auto.ConfigValue{Value: armInstanceType},
+		"ddinfra:aws/defaultInstanceType":    auto.ConfigValue{Value: x86InstanceType},
+		"ddinfra:aws/defaultKeyPairName":     auto.ConfigValue{Value: "datadog-agent-kitchen"},
+		"ddinfra:aws/defaultPrivateKeyPath":  auto.ConfigValue{Value: SSHKeyFile},
+		"ddinfra:aws/defaultSecurityGroups":  auto.ConfigValue{Value: securityGroups},
+		"ddinfra:aws/defaultSubnets":         auto.ConfigValue{Value: subnets},
+		"microvm:microVMConfigFile":          auto.ConfigValue{Value: vmConfig},
+		"microvm:libvirtSSHKeyFileX86":       auto.ConfigValue{Value: sshKeyX86},
+		"microvm:libvirtSSHKeyFileArm":       auto.ConfigValue{Value: sshKeyArm},
+	}
+
+	upResult, err := stackManager.GetStack(systemProbeTestEnv.context, systemProbeTestEnv.envName, systemProbeTestEnv.name, config, func(ctx *pulumi.Context) error {
+		awsEnvironment, err := aws.NewEnvironment(ctx)
+		if err != nil {
+			return err
```

Comment:
suggestion: wrap your errors with `fmt.Errorf` and add some context. For example: `return fmt.Errorf("aws new environment: %w", err)`. That way, even obtuse errors have some context info about where they came from.